primitives/src/pcs/multilinear_kzg/mod.rs

```diff
@@ -28,9 +28,9 @@ use ark_std::{
     end_timer, format,
     marker::PhantomData,
     rand::{CryptoRng, RngCore},
+    rc::Rc,
```

Comment:
All of this is guaranteed to only be accessed within a single thread?

Reply:
Okay, never mind, I get it. This stems from `merge_polynomials` taking `&[Self::Polynomial]`, and Rust not really having a good way to gracefully handle `&[&T]` for reentrant operations.

Support for variadic generics would be very nice here. 😒 

It would seem `Rc` is, in fact, correct.

Reply:
I have to switch to this to compile due to missing `CanonicalSerde` mentioned here: https://github.com/arkworks-rs/algebra/issues/649.

It's a temporary (and wrong) decision to use `Rc`, `Arc` doesn't have CanonicalSerde in `no_std` mode at all, so I will have to remove the trait bounds in `pcs/mod.rs::trait PolynomicalCommitmentScheme::Polynomial`; but for `Rc` only the CanonicalDeserialize is missing, but at least it's available in `no_std` mode, thus I can still compile it.

For WASM env, the "multi-threading" seems to only support message-passing concurrency, but not shared reference concurrency, I need to change this code, not even `Rc` would work.

Reply:
> This stems from merge_polynomials taking &[Self::Polynomial], and Rust not really having a good way to gracefully handle &[&T] for reentrant operations.

the actual reason was simpler than this, your point is something I didn't consider btw.
Originally we use `Arc` and it was fine (on non-wasm platforms). I'm not sure about the "reentrant op" you mentioned. 

Now, I'm leaning towards, in `no_std` I just pass in owned value itself, whereas otherwise, I'll just use `Arc`

Reply:
see if you like [d1e8c3b](https://github.com/EspressoSystems/jellyfish/pull/254/commits/d1e8c3b1667052923932283cb68a3788c7252c42) @nyospe

Reply:
That'll work. Somewhere in the back of my brain, I think I have the glimmering of a better solution, but it's being stubborn, and this works for now, at the cost of some potentially avoidable copies.

The _correct_ solution is a variadic function, but in Rust, that requires a macro, and that's a PitA, and borderline impossible in a trait method, so let's go with this.